### DISCUSSION TITLE
How to cyclize a linear peptide composed of natural and non-natural residues?

### DISCUSSION THREAD

**Discussion by hahadg** [rdkit/rdkit#8079] [Q&A]
Hi everyone,

There are some linear peptides which are composed of natural and non-natural amino acids, and I want to link their N-terminal and C-terminal to form cyclic peptides. I tried to use reaction SMARTS to form intra-molecular peptide bond by rdkit. Here is an attempt:
```
from rdkit import Chem
from rdkit.Chem import AllChem

rxn_cyc_CaCa = AllChem.ReactionFromSmarts(
                           '([$([CX3](=[OX1])[CX4][NX3]):1]-[OD1:2].[$([N&H2,N&R&!H0][CX4][CX3](=[OX1])):3])>>'
                           '[CX3:1]-[NX3:3].[OD0:2]')
liner_pep = Chem.MolFromSmiles('NC(C)C(NC(Cc1c2ccc(N)cc2[nH]c1)C(NC(CCCNC(N)=N)C(NCC(CS)C(NCC(NC(Cc3ccc(O)cc3C(O)=O)C(N4CCC(C(NC(C(NC(Cc5c[nH]cn5)C(O)=O)=O)C(C)CCI)=O)C4)=O)=O)=O)=O)=O)=O')
rnts = ((liner_pep, ))
cyc_peps = rxn_cyc_CaCa.RunReactants(rnts)
print('Num. of products: ' + len(cyc_peps))
print(Chem.MolToSmiles(cyc_peps[0][0]))
```
input:
![image](https://github.com/user-attachments/assets/b5d9bcd6-5c28-4c8a-90c4-155a40523a99)

output: 
``1
CC1NC(=O)C(Cc2c[nH]cn2)NC(=O)C(C(C)CCI)NC(=O)C2CCN(C2)C(=O)C(Cc2ccc(O)cc2C(=O)O)NC(=O)CNC(=O)C(CS)CNC(=O)C(CCCNC(=N)N)NC(=O)C(Cc2c[nH]c3cc(N)ccc23)NC1=O``
![image](https://github.com/user-attachments/assets/88334669-c84c-47d0-9f50-129e0b64ae59)

The product is unique and correct. However, when the N-termi is methylated (as follows), the above ``rxn_cyc_CaCa`` is no longer suitable.
methylated N-terminal:
``ICCC(C)C(C(NC(Cc1c[nH]cn1)C(O)=O)=O)NC(C(C2)CCN2C(C(Cc3ccc(O)cc3C(O)=O)NC(CNC(C(CS)CNC(C(CCCNC(N)=N)NC(C(Cc4c5ccc(N)cc5[nH]c4)NC(C(C)NC)=O)=O)=O)=O)=O)=O)=O``
![image](https://github.com/user-attachments/assets/5edc178d-3bc9-404e-a6cd-974e84dbeaeb)

 Thus, I changed the reaction SMARTS to:
```
rxn_cyc_CaCa = AllChem.ReactionFromSmarts(
                           '([$([CX3](=[OX1])[CX4][NX3]):1]-[OD1:2].[$([N&H2,N&R&!H0,N&D2&!H0][CX4][CX3](=[OX1])):3])>>'
                           '[CX3:1]-[NX3:3].[OD0:2]')
```
output: 
``7
CNC(C)C(=O)NC(Cc1c[nH]c2cc(N)ccc12)C(=O)NC(CCCNC(=N)N)C(=O)NCC(CS)C(=O)NCC(=O)NC(Cc1ccc(O)cc1C(=O)O)C(=O)N1CCC(C(=O)NC(C(=O)N2C(=O)C2Cc2c[nH]cn2)C(C)CCI)C1``
![image](https://github.com/user-attachments/assets/3eadc267-8a14-4627-a6c9-b32cbc0f5497)

No doubts, it gave multiple products (7) and some products are produced by coupling -COOH with the NH in the middle peptide bond.  It is not desired. **Is there any way to distinguish the NH in the N-terminal (the blue one) with the NH in the middle (the red one)** _?_ 
![image](https://github.com/user-attachments/assets/b9687616-a1c6-4668-9f66-8ef75bf28108)

Many thanks for your help!

**Comment by greglandrum** [comment]
Hi @Nicole-DH,
I think you can capture the terminal methylated N by allowing a bond to an N which has a single H and a CH3 group:
```
rxn_cyc_CaCa_2 = AllChem.ReactionFromSmarts(
                           '([$([CX3](=[OX1])[CX4][NX3]):1]-[OD1:2].[$([N&H2,N&R&!H0,NH&$(N[CH3])][CX4][CX3](=[OX1])):3])>>'
                           '[CX3:1]-[NX3:3].[OD0:2]')


pep2 = Chem.MolFromSmiles('ICCC(C)C(C(NC(Cc1c[nH]cn1)C(O)=O)=O)NC(C(C2)CCN2C(C(Cc3ccc(O)cc3C(O)=O)NC(CNC(C(CS)CNC(C(CCCNC(N)=N)NC(C(Cc4c5ccc(N)cc5[nH]c4)NC(C(C)NC)=O)=O)=O)=O)=O)=O)=O')
cyc_peps = rxn_cyc_CaCa_2.RunReactants((pep2,))
print('Num. of products:',len(cyc_peps))
print(Chem.MolToSmiles(cyc_peps[0][0]))
```
This produces the molecule:
![image](https://github.com/user-attachments/assets/2ab9d129-21fd-4085-894f-fba71f254a7b)

which I think is what you were looking for.
